src/apps/btc/btc_sign.c

```diff
@@ -384,27 +446,36 @@ app_btc_sign_error_t app_btc_sign_output(
                 return _error(APP_BTC_SIGN_ERR_UNKNOWN);
             }
 
-            size_t out_size = 0;
             // construct pkScript
             if (!btc_common_outputhash_from_pubkeyhash(
                     _init_request.script_config.config.simple_type,
                     pubkey_hash160,
-                    hash.bytes,
-                    &out_size)) {
+                    hash_bytes,
+                    &hash_size)) {
                 return _error(APP_BTC_SIGN_ERR_UNKNOWN);
             }
             output_type =
                 btc_common_determine_output_type(_init_request.script_config.config.simple_type);
-
-            hash.size = (pb_size_t)out_size;
             break;
         }
+        case BTCScriptConfig_multisig_tag:
+            if (!btc_common_outputhash_from_multisig_p2wsh(
+                    &_init_request.script_config.config.multisig,
+                    request->keypath[request->keypath_count - 2],
+                    request->keypath[request->keypath_count - 1],
+                    hash_bytes)) {
+                return _error(APP_BTC_SIGN_ERR_UNKNOWN);
+            }
+            hash_size = 32;
+            output_type = BTCOutputType_P2WSH;
+            break;
         default:
             return _error(APP_BTC_SIGN_ERR_INVALID_INPUT);
         }
 
     } else {
-        hash = request->hash;
+        memcpy(hash_bytes, request->hash.bytes, sizeof(hash_bytes));
```

Comment:
`sizeof(has_bytes)` should be `min(sizeof(hash_bytes), request->hash.size)` Or there should be an abort in case they aren't the same.

Reply:
copying only request->hash.size now, as that is either 20 or 32.

for the record, the previous way (copying 32 bytes) works just as well, as both buffers have the same capacity (32 bytes), so copying all of it does not technically hurt.

Reply:
No, if you don't use `min(..)` or verify that they are the same you will either have an out-of-bounds write or an out-of-bounds read. Both are equally bad.

Reply:
> copying only request->hash.size now, since min(sizeof(hash_bytes), request->hash.size) is request->hash.size always.

Adding the below code will be optimized away in case they are the same. It is just a sanity check for ourselves.

```
if(request->hash != sizeof(hash_bytes) {
    Abort(...)'
}
```

Reply:
since the last push `hash_bytes` is initialized with `sizeof(request->hash.bytes)` size, so it is the same.

Reply:
Ah OK, I'll rereview then